pkg/endpoint/types.go

```diff
@@ -32,6 +32,8 @@ type Provider interface {
 type Endpoint struct {
 	net.IP `json:"ip"`
 	Port   `json:"port"`
+	Weight `json:"weight"`
+	Name   string `json:"name"`
```

Comment:
Please add a comment for the field here. It's not clear what `Name` implies here. It seems to be used as an endpoint's zone instead, so maybe just call this `Zone`.

Reply:
`Name` is the name of the endpoint, which is optional. In OSM multicluster mode, endpoints from local cluster are not assigned names, while endpoints from remote cluster have names. Each remote cluster has one endpoint, which is the one of its multicluster gateway. 

I think keeping it `Name` is better, as it's more general than `Zone`.

Reply:
I think making the struct field names specific makes it easier to understanding. I wouldn't generalize this unless there is a reason to, because it can be unnecessarily confusing. `Name` for an endpoint seems ambiguous to me, because an endpoint is just a representation of an ip:port combination, and naming this seems weird. On the other hand, an endpoint being a part of a `zone` (or availabilityZone, localityZone etc.) is much more clear and self-explanatory.